### ISSUE TITLE
Promise.withResolvers is not detected when targeting to safari 17.3 and baseline 2023

### ISSUE THREAD

**Issue by fungcheukyin-TeamLab** [amilajack/eslint-plugin-compat#669]
My project is having an issue that Promise.withResolvers is used in our codes and EsLint does not raise any error against it even we are setting the browserslist as "iOS >= 17.0" or "baseline 2023"
```
"browserslist": [
    "extend browserslist-config-baseline"
  ],
  "browserslist-config-baseline": {
    "logConfigToConsole": true,
    "widelyAvailableOnDate": "2023-01-01"
  },
```

Promise.withResolvers is added from 2024.01 and https://caniuse.com/?search=Promise.withResolvers stated that it is available from iOS safari 17.4. It is also in the Baseline 2024  as Newly Available.

I tried to dick into the codebase and found that Promise.withResolvers is not included in caniuse-provider.
https://github.com/amilajack/eslint-plugin-compat/blob/v4.2.0/src/providers/caniuse-provider.ts

Is there any plan to fix this issue?
